plugins/search-react/src/components/SearchResultLimiter/SearchResultLimiter.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactNode, ChangeEvent, useCallback } from 'react';
+import {
+  Box,
+  InputBase,
+  MenuItem,
+  Select,
+  Typography,
+  useTheme,
+} from '@material-ui/core';
+import { useSearch } from '../../context';
+
+/**
+ * Props for {@link SearchResultLimiterBase}.
+ * @public
+ */
+export type SearchResultLimiterBaseProps = {
+  id?: string;
+  className?: string;
+  /**
+   * A label for the combobox.
+   */
+  label?: ReactNode;
+  /**
+   * The combobox labels, defaults to 10, 25, 50 and 100.
+   */
+  options?: number[];
```

Comment:
should we update this to make sure options over 100 is not added?

Reply:
Good idea, will do!

Reply:
closing this comment after creating a separate pull request for the Frontend:
https://github.com/backstage/backstage/pull/13968